test/metabase/stale_test.clj

```diff
@@ -0,0 +1,40 @@
+(ns metabase.stale-test
+  (:require [metabase.test :as mt])
+  (:import (java.time LocalDate LocalDateTime)))
+
+(set! *warn-on-reflection* true)
+
+(defn date-months-ago
+  "Get the date n months ago."
+  ^LocalDate
+  [n]
+  (-> (LocalDate/now)
+      (.minusMonths n)))
+
+(defn datetime-months-ago
+  "Get the datetime n months ago."
+  ^LocalDateTime
+  [n]
+  (-> (LocalDateTime/now)
+      (.minusMonths n)))
+
+(defn stale-dashboard [dashboard]
+  ;; assoc the dashboard with the current time minus 7 months
+  (assoc dashboard :last_viewed_at (datetime-months-ago 7)))
+
+(defn stale-card [card]
+  ;; assoc the card with the current time minus 7 months
+  (assoc card :last_used_at (datetime-months-ago 7)))
+
+(defmacro with-stale-items [inputs & body]
```

Comment:
If I'm reading this right, it will only work with cards and dashboards, do you think its worth failing outright when a model that won't be marked as stale is included?

Reply:
Yeah, that's probably a good idea.